### PR TITLE
BRD - UI: Feat - Manage nr modal functionality

### PR DESCRIPTION
**Pull request by deetz99**
Partial: bcGov/entity#22573

- add functionality to manage name request modal

### CODE REVIEW

File: business-registry-dashboard/app/components/Form/AddNameRequest/Base.vue
Comment (deetz99) [inline business-registry-dashboard/app/components/Form/AddNameRequest/Base.vue:1]: - pass error context to nameRequestError emitted event
- use fetch request for affiliation from composable
- add screen reader alert text for form validation, might need to expand on this

File: business-registry-dashboard/app/components/Form/AddNameRequest/Error.vue
Comment (deetz99) [inline business-registry-dashboard/app/components/Form/AddNameRequest/Error.vue:1]: - display different error text based on error object
- add screen reader error alert text

File: business-registry-dashboard/app/components/Form/AddNameRequest/index.vue
Comment (deetz99) [inline business-registry-dashboard/app/components/Form/AddNameRequest/index.vue:1]: - update so I can pass the error from the form component to the error component

File: business-registry-dashboard/app/components/Modal/Base.vue
Comment (deetz99) [inline business-registry-dashboard/app/components/Modal/Base.vue:1]: update base modal to be able to display arbitrary error state

File: business-registry-dashboard/app/composables/useAffiliations.ts
Comment (deetz99) [inline business-registry-dashboard/app/composables/useAffiliations.ts:1]: - add nr affiliations util/fetch request implementation

File: business-registry-dashboard/app/composables/useBrdModals.ts
Comment (deetz99) [inline business-registry-dashboard/app/composables/useBrdModals.ts:1]: - useModal only handles one modal at a time so moved to top scope and made general close() function
- add function for showing general nr error modal (used if theres an error trying to add a nr as a staff)

File: business-registry-dashboard/app/pages/index.vue
Comment (deetz99) [inline business-registry-dashboard/app/pages/index.vue:1]: - move logic into composable
- had to wrap table in client only to use the loading state correctly

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-registry-dev--pr-19-nritl3b0.web.app

**Review by thorwolpert — APPROVED** [review]

**Pull request merged**